src/test/java/mate/academy/bookshop/service/impl/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,170 @@
+
+    private UserRegistrationRequestDto requestDto;
+    private User user;
+    private UserResponseDto userResponseDto;
+    private ShoppingCart shoppingCart;
+
+    @BeforeEach
+    void setUp() {
+        requestDto = new UserRegistrationRequestDto();
+        requestDto.setEmail("user@example.com");
+        requestDto.setPassword("password");
+
+        user = new User();
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(requestDto.getPassword());
+
+        userResponseDto = new UserResponseDto();
+        userResponseDto.setEmail(user.getEmail());
+
+        shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+    }
+
+    @DisplayName("Register new user with valid credentials")
+    @Test
+    void register_UserWithUniqueEmail_returnsUserResponseDto() throws RegistrationException {
+        when(userRepository.existsByEmail(requestDto.getEmail())).thenReturn(false);
+        when(userMapper.toModel(requestDto)).thenReturn(user);
+        when(passwordEncoder.encode(user.getPassword())).thenReturn("encodedPassword");
+        when(roleRepository.findByName(RoleName.ROLE_USER)).thenReturn(Set.of(new Role()));
+        when(userRepository.save(user)).thenReturn(user);
+        when(userMapper.toDto(user)).thenReturn(userResponseDto);
+
+        UserResponseDto response = authenticationService.register(requestDto);
+
+        assertEquals(userResponseDto, response);
+    }
+
+    @DisplayName("Register new user with invalid credentials")
+    @Test
+    void register_UserWithNotUniqueEmail_ThrowsRegistrationException()
+            throws RegistrationException {
+        when(userRepository.existsByEmail(requestDto.getEmail())).thenReturn(true);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, ()
+                -> authenticationService.register(requestDto));
+
+        String expected = "There is another user with email " + requestDto.getEmail();
+        String actual = exception.getMessage();
+        assertEquals(expected, actual);
+    }
+
+    @DisplayName("Roles are assigned correctly")
+    @Test
+    void register_RolesAreAssigned_True() throws RegistrationException {
+        Role role = new Role();
+        role.setName(RoleName.ROLE_USER);
+
+        when(userRepository.existsByEmail(requestDto.getEmail())).thenReturn(false);
+        when(userMapper.toModel(requestDto)).thenReturn(user);
+        when(passwordEncoder.encode(user.getPassword())).thenReturn("encodedPassword");
+        when(roleRepository.findByName(RoleName.ROLE_USER)).thenReturn(Set.of(new Role()));
+        when(userRepository.save(user)).thenReturn(user);
+
+        authenticationService.register(requestDto);
+
+        verify(roleRepository).findByName(RoleName.ROLE_USER);
+        verify(userRepository).save(user);
+
+        assertTrue(user.getRoles().size() == 1);
+        assertTrue(user.getRoles().contains(role));
+    }
```

Comment:
No reason to this test, you should have 1 test with successuff case where role of user is checked